core/bin/snapshots_creator/src/tests.rs

```diff
@@ -0,0 +1,212 @@
+        l1_tx_count: 0,
+        l2_tx_count: 0,
+        base_fee_per_gas: 0,
+        l1_gas_price: 0,
+        l2_fair_gas_price: 0,
+        base_system_contracts_hashes: Default::default(),
+        protocol_version: Some(Default::default()),
+        virtual_blocks: 0,
+    };
+
+    conn.blocks_dal()
+        .insert_miniblock(&miniblock_header)
+        .await
+        .unwrap();
+    conn.storage_logs_dal()
+        .insert_storage_logs(miniblock_number, &[(H256::zero(), block_logs)])
+        .await;
+}
+
+async fn create_l1_batch(
+    conn: &mut StorageProcessor<'_>,
+    l1_batch_number: L1BatchNumber,
+    logs_for_initial_writes: &[StorageLog],
+) {
+    let mut header = L1BatchHeader::new(
+        l1_batch_number,
+        0,
+        Address::default(),
+        Default::default(),
+        Default::default(),
+    );
+    header.is_finished = true;
+    conn.blocks_dal()
+        .insert_l1_batch(&header, &[], BlockGasCount::default(), &[], &[])
+        .await
+        .unwrap();
+    conn.blocks_dal()
+        .mark_miniblocks_as_executed_in_l1_batch(l1_batch_number)
+        .await
+        .unwrap();
+
+    let mut written_keys: Vec<_> = logs_for_initial_writes.iter().map(|log| log.key).collect();
+    written_keys.sort_unstable();
+    conn.storage_logs_dedup_dal()
+        .insert_initial_writes(l1_batch_number, &written_keys)
+        .await;
+}
+
+async fn prepare_postgres(
+    rng: &mut impl Rng,
+    conn: &mut StorageProcessor<'_>,
+    block_count: u32,
+) -> ExpectedOutputs {
+    conn.protocol_versions_dal()
+        .save_protocol_version_with_tx(ProtocolVersion::default())
+        .await;
+
+    let mut outputs = ExpectedOutputs::default();
+    for block_number in 0..block_count {
+        let logs = gen_storage_logs(rng, 100);
+        create_miniblock(conn, MiniblockNumber(block_number), logs.clone()).await;
+
+        let factory_deps = gen_factory_deps(rng, 10);
+        conn.storage_dal()
+            .insert_factory_deps(MiniblockNumber(block_number), &factory_deps)
+            .await;
+
+        // Since we generate `logs` randomly, all of them are written the first time.
+        create_l1_batch(conn, L1BatchNumber(block_number), &logs).await;
+
+        if block_number + 1 < block_count {
+            outputs.deps.extend(factory_deps.into_values());
+            let logs = logs.into_iter().map(|log| (log.key, log.value));
+            outputs.storage_logs.extend(logs);
+        }
+    }
+    outputs
+}
+
+#[tokio::test]
+async fn basic_workflow() {
+    let pool = ConnectionPool::test_pool().await;
+    let mut rng = thread_rng();
+    let object_store_factory = ObjectStoreFactory::mock();
+    let object_store = object_store_factory.create_store().await;
+
+    // Insert some data to Postgres.
+    let mut conn = pool.access_storage().await.unwrap();
+    let expected_outputs = prepare_postgres(&mut rng, &mut conn, 10).await;
+    drop(conn);
```

Comment:
why do you need to drop conn here?

Reply:
Sorry; just got used to the previous implementation of a test DB connection pool that didn't allow to acquire more than one connection at a time 🙃